pkg/collector/scheduler.go

```diff
@@ -190,6 +191,45 @@ func (s *CheckScheduler) getChecks(config integration.Config) ([]check.Check, er
 			log.Debugf("Loading check instance for check '%s' using default loaders", config.Name)
 		}
 
+		// Special case to use Core loader by default for SNMP integration
+		if config.Name == "snmp" && selectedInstanceLoader == "" {
+			var coreLoader check.Loader
+			var pythonLoader check.Loader
+
+			for _, loader := range s.loaders {
+				if loader.Name() == "core" {
+					coreLoader = loader
+				}
+				if loader.Name() == "python" {
+					pythonLoader = loader
+				}
+			}
+
+			if coreLoader != nil && pythonLoader != nil {
+				coreCheck, coreErr := coreLoader.Load(s.senderManager, config, instance)
+				pythonCheck, pythonErr := pythonLoader.Load(s.senderManager, config, instance)
+
+				if coreErr != nil {
+					if pythonErr != nil {
+						errors = append(errors,
+							fmt.Sprintf("%v: %s", pythonLoader, pythonErr),
+							fmt.Sprintf("%v: %s", coreLoader, coreErr))
+						log.Errorf("Unable to load a check from instance of config '%s': %s", config.Name, strings.Join(errors, "; "))
+					} else {
+						checks = append(checks, pythonCheck)
+					}
+					continue
+				}
+
+				if pythonErr == nil {
+					coreCheck.(*snmp.Check).SetPythonCheck(pythonCheck)
+				}
```

Comment:
I'm not sure that's the best way to do this, but anyway if you're going this way then you should avoid importing the implementation of a check, and just cast to an interface instead, eg.:
```suggestion
				if pythonErr == nil {
					coreCheck.(interface { SetPythonCheck(check.Check) }).SetPythonCheck(pythonCheck)
				}
```
This way you don't import the impl.

(disclaimer I haven't tested this, it might not be quite the correct syntax)